### ISSUE TITLE
Dropdown: expanded/collapsed state is not announced on MacOS + Safari

### ISSUE THREAD

**Issue by nikitaorliak-cengage** [cengage/react-magma#2563]
**Describe the bug**
When navigating to the dropdown button, NVDA + Chrome announces the 'collapsed' state and, after opening, announces 'expanded'. However, Safari + VoiceOver does not announce the expanded/collapsed state.

**To Reproduce**
1. MacOS + Safari (check others too)
2. Open Storybook/Docs
3. `Dropdown` component
4. Turn on NVDA/VO
5. Open the Dropdown
6. Notice that expanded/collapsed state is not announced

**Expected behavior**
expanded/collapsed state should be announced on every device/browser

**Additional context**
- It should be fixed on 2.6.2 version too
- Slack conversation: https://cengage.slack.com/archives/C8Y4FR63U/p1782388608717709
